docs/feature-variables-and-me.md

```diff
@@ -0,0 +1,266 @@
+
+For example on Android: `getDrawable("settings-menu-item-icon")` may get a string `"ic_settings"`, which is then resolved to `R.drawable.ic_settings`, which is then resolved to `context.resources.getDrawable(R.drawable.ic_settings)`.
+
+On iOS: `getImage("settings-menu-item-icon")` may get a string `"icon_photon_gear"`, which is then used to get the named `UIImage` with `UIImage(named:in:)`.
+
+## Making JSON more manageable
+
+We focused on the settings menu item in the above example, as a way of making a small enough example to reason about in this documentation, but it made for some very long variable names. The `Variables` object has itself a `getVariables(key: String)` method to make navigating the JSON more easily. This in turn allows the JSON to be organized in different ways.
+
+```json
+{
+    "settings": {
+        "icon": "ic_settings",
+        "title": "Settings",
+        "action": "firefox://settings",
+        "enabled": true
+    },
+    "bookmarks": {
+        "icon": "ic_bookmarks",
+        "title": "View Bookmarks",
+        "action": "firefox://bookmark_list",
+        "enabled": true
+    },
+    "history": {
+        "icon": "ic_history",
+        "title": "View History",
+        "action": "firefox://history_list",
+        "enabled": true
+    }
+}
+```
+
+This might be accessed in Kotlin with:
+
+```kotlin
+val menuVariables = nimbus.getVariables("app-menu")
+var settingsItem = menuVariables.getVariables("settings").let { vars ->
+    val action: String = vars?.getString("action") ?: "firefox://settings"
+    val title: String = vars?.getText("title") ?: context.getString(R.string.app_menu_settings_title)
+    val icon: Drawable = vars?.getDrawable("icon") ?: context.getDrawable(R.drawable.ic_settings)
+
+    MenuItem(icon, title, action)
+}
+```
+
+In Swift:
+
+```swift
+let menuVariables = nimbus.getVariables("app-menu")
+let settingsItem = menuVariables.getVariables("settings").let { vars ->
+    let action: String = vars?.getString("action") ?? "firefox://settings"
+    let title: String = vars?.getText("title") ?? Strings.AppMenuSettingsTitle
+    let icon: UIImage = vars?.getImage("icon") ?? UIImage(named: "icon-photon-gear")
+    let isEnabled: Bool = vars?.getBool("enabled") ?? true
+
+    MenuItem(icon, title, action)
+}
+```
+
+> 👋 Information
+>
+> `variables.getVariables()` can be arbitrarily deep. `variables.getVariables()` returns an optional `Variables` object.
+
+## Recording exposure events
+
+> ⛅️🔬🔭 Enrollment versus Exposure
+>
+> When a client is selected to take part in an experiment, they are _assigned_ a branch. This is _enrollment_.
+> However, the user may not be _exposed_ to the branch until sometime later. The exposure is the earliest moment that the user could be affected by the experimental treatment.
+>
+> Nimbus tracks the enrollment and the exposure events using Glean.
+
+For experiments in Firefox for iOS and Android, enrollment happens shortly after app-startup.
```

Comment:
Do enrollment and exposure only happen once for each user? (per experiment)

Reply:
Added an additional sentence to the concept-box above.